main.py

```diff
@@ -2,21 +2,26 @@
 import numpy as np
 from numpy import asarray
 from PIL import Image
+import random,csv
 
 # Freddie ;)
 """
 Converts a CSV of image pixels into an array of pixel values
 INPUTS: CSV File name
 RETURNS: 2d array of all the pixel values for each image [[pixels in img 1], [pixels in img 2], ..]
 """
-def CsvToArray():
-  sigmalabels = []
+def CsvToArray(filename="train.csv"):
   data = []
-  with open("train.csv", "r") as file: # Might be good to make the file name a param, and default it to the "train.csv" (I belive there's a programming principle which involves this idea, but I can't remember it!)
-    training_data = file.reader(file)
-    for row in data:
-      sigmalabels.append(row[0])
-      data.append(row[1:])
+  with open(filename, "r") as file: # Might be good to make the file name a param, and default it to the "train.csv" (I belive there's a programming principle which involves this idea, but I can't remember it!)
+    training_data = csv.reader(file)
+    counter = 0 
+    for row in training_data:
```

Comment:
Could even skip the selection statement every time, by popping the first element of the array before the for loop!
```suggestion
    sigmalabels = training_data.pop(0)
    for row in training_data:
     ....
```